source/code/shared/tools/scx_ssl_config/scxsslcert.cpp

```diff
@@ -370,13 +379,41 @@ void SCXSSLCertificate::DoGenerate()
         }
 
         {
-            RSA * rsa = RSA_generate_key(newKeyLength, 0x10001, 0, 0);
-            if ( ! rsa )
-            {
-                throw SCXCoreLib::SCXNULLPointerException(L"Error allocating RSA structure.",
-                                                      SCXSRCLOCATION);
-            }
-            if ( ! EVP_PKEY_assign_RSA(pkey.Get(), rsa))
+            int ret = 1;
+
+            #if OPENSSL_VERSION_NUMBER < 0x0090800fL // SSL version lower than 0.9.8? It is needed for Solaris-10.
+                RSA * rsa = RSA_generate_key(newKeyLength, 0x10001, 0, 0);
+
+                if ( ! rsa )
+                {
+                    throw SCXCoreLib::SCXNULLPointerException(L"Error allocating RSA structure.",
+                                                          SCXSRCLOCATION);
+                }
+
+            #else
+
+                //RSA * rsa = RSA_generate_key(newKeyLength, 0x10001, 0, 0);
```

Comment:
This can be deleted, it's clear what's going on without this stale line.

Reply:
Will remove this line.

Reply:
Fixed